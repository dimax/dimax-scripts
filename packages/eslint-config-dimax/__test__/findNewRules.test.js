/**
 * @author @lgraziani2712
 *
 * @flow
 */

'use strict';

const getRuleFinder = require('eslint-find-rules');

const ruleFinder = getRuleFinder('./packages/eslint-config-dimax/index.js');

test('Control eslint used rules', () => {
  expect(ruleFinder.getCurrentRules()).toMatchSnapshot();
});
