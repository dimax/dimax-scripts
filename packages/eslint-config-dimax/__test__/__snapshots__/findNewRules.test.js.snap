// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Control eslint used rules 1`] = `
Array [
  "accessor-pairs",
  "array-bracket-newline",
  "array-bracket-spacing",
  "array-callback-return",
  "array-element-newline",
  "arrow-body-style",
  "arrow-parens",
  "arrow-spacing",
  "block-scoped-var",
  "block-spacing",
  "brace-style",
  "callback-return",
  "camelcase",
  "capitalized-comments",
  "class-methods-use-this",
  "comma-dangle",
  "comma-spacing",
  "comma-style",
  "complexity",
  "computed-property-spacing",
  "consistent-return",
  "consistent-this",
  "constructor-super",
  "curly",
  "default-case",
  "dot-location",
  "dot-notation",
  "eol-last",
  "eqeqeq",
  "filenames/match-exported",
  "flowtype/boolean-style",
  "flowtype/define-flow-type",
  "flowtype/delimiter-dangle",
  "flowtype/generic-spacing",
  "flowtype/no-types-missing-file-annotation",
  "flowtype/no-weak-types",
  "flowtype/require-parameter-type",
  "flowtype/require-return-type",
  "flowtype/require-valid-file-annotation",
  "flowtype/semi",
  "flowtype/space-after-type-colon",
  "flowtype/space-before-generic-bracket",
  "flowtype/space-before-type-colon",
  "flowtype/type-id-match",
  "flowtype/union-intersection-spacing",
  "flowtype/use-flow-type",
  "flowtype/valid-syntax",
  "for-direction",
  "func-call-spacing",
  "func-name-matching",
  "func-names",
  "func-style",
  "function-paren-newline",
  "generator-star-spacing",
  "getter-return",
  "global-require",
  "guard-for-in",
  "handle-callback-err",
  "id-blacklist",
  "id-length",
  "id-match",
  "implicit-arrow-linebreak",
  "import/default",
  "import/export",
  "import/extensions",
  "import/imports-first",
  "import/named",
  "import/namespace",
  "import/newline-after-import",
  "import/no-anonymous-default-export",
  "import/no-deprecated",
  "import/no-duplicates",
  "import/no-mutable-exports",
  "import/no-named-as-default",
  "import/no-named-as-default-member",
  "import/no-unresolved",
  "import/order",
  "import/prefer-default-export",
  "indent",
  "indent-legacy",
  "init-declarations",
  "jest/no-disabled-tests",
  "jest/no-focused-tests",
  "jest/no-identical-title",
  "jest/valid-expect",
  "jsx-quotes",
  "key-spacing",
  "keyword-spacing",
  "line-comment-position",
  "linebreak-style",
  "lines-around-comment",
  "lines-around-directive",
  "lines-between-class-members",
  "max-depth",
  "max-len",
  "max-lines",
  "max-nested-callbacks",
  "max-params",
  "max-statements",
  "max-statements-per-line",
  "multiline-comment-style",
  "multiline-ternary",
  "new-cap",
  "new-parens",
  "newline-after-var",
  "newline-before-return",
  "newline-per-chained-call",
  "no-alert",
  "no-array-constructor",
  "no-await-in-loop",
  "no-bitwise",
  "no-buffer-constructor",
  "no-caller",
  "no-case-declarations",
  "no-catch-shadow",
  "no-class-assign",
  "no-compare-neg-zero",
  "no-cond-assign",
  "no-confusing-arrow",
  "no-console",
  "no-const-assign",
  "no-constant-condition",
  "no-continue",
  "no-control-regex",
  "no-debugger",
  "no-delete-var",
  "no-div-regex",
  "no-dupe-args",
  "no-dupe-class-members",
  "no-dupe-keys",
  "no-duplicate-case",
  "no-duplicate-imports",
  "no-else-return",
  "no-empty",
  "no-empty-character-class",
  "no-empty-function",
  "no-empty-pattern",
  "no-eq-null",
  "no-eval",
  "no-ex-assign",
  "no-extend-native",
  "no-extra-bind",
  "no-extra-boolean-cast",
  "no-extra-label",
  "no-extra-parens",
  "no-extra-semi",
  "no-fallthrough",
  "no-floating-decimal",
  "no-func-assign",
  "no-global-assign",
  "no-implicit-coercion",
  "no-implicit-globals",
  "no-implied-eval",
  "no-inline-comments",
  "no-inner-declarations",
  "no-invalid-regexp",
  "no-invalid-this",
  "no-irregular-whitespace",
  "no-iterator",
  "no-label-var",
  "no-labels",
  "no-lone-blocks",
  "no-lonely-if",
  "no-loop-func",
  "no-magic-numbers",
  "no-mixed-operators",
  "no-mixed-requires",
  "no-mixed-spaces-and-tabs",
  "no-multi-assign",
  "no-multi-spaces",
  "no-multi-str",
  "no-multiple-empty-lines",
  "no-native-reassign",
  "no-negated-condition",
  "no-negated-in-lhs",
  "no-nested-ternary",
  "no-new",
  "no-new-func",
  "no-new-object",
  "no-new-require",
  "no-new-symbol",
  "no-new-wrappers",
  "no-obj-calls",
  "no-octal",
  "no-octal-escape",
  "no-param-reassign",
  "no-path-concat",
  "no-plusplus",
  "no-process-env",
  "no-process-exit",
  "no-proto",
  "no-prototype-builtins",
  "no-redeclare",
  "no-regex-spaces",
  "no-restricted-globals",
  "no-restricted-imports",
  "no-restricted-modules",
  "no-restricted-properties",
  "no-restricted-syntax",
  "no-return-assign",
  "no-return-await",
  "no-script-url",
  "no-self-assign",
  "no-self-compare",
  "no-sequences",
  "no-shadow",
  "no-shadow-restricted-names",
  "no-spaced-func",
  "no-sparse-arrays",
  "no-sync",
  "no-tabs",
  "no-template-curly-in-string",
  "no-ternary",
  "no-this-before-super",
  "no-throw-literal",
  "no-trailing-spaces",
  "no-undef",
  "no-undef-init",
  "no-undefined",
  "no-underscore-dangle",
  "no-unexpected-multiline",
  "no-unmodified-loop-condition",
  "no-unneeded-ternary",
  "no-unreachable",
  "no-unsafe-finally",
  "no-unsafe-negation",
  "no-unused-expressions",
  "no-unused-labels",
  "no-unused-vars",
  "no-use-before-define",
  "no-useless-call",
  "no-useless-computed-key",
  "no-useless-concat",
  "no-useless-constructor",
  "no-useless-escape",
  "no-useless-rename",
  "no-useless-return",
  "no-var",
  "no-void",
  "no-warning-comments",
  "no-whitespace-before-property",
  "no-with",
  "nonblock-statement-body-position",
  "object-curly-newline",
  "object-curly-spacing",
  "object-property-newline",
  "object-shorthand",
  "one-var",
  "one-var-declaration-per-line",
  "operator-assignment",
  "operator-linebreak",
  "padded-blocks",
  "padding-line-between-statements",
  "prefer-arrow-callback",
  "prefer-const",
  "prefer-destructuring",
  "prefer-numeric-literals",
  "prefer-promise-reject-errors",
  "prefer-reflect",
  "prefer-rest-params",
  "prefer-spread",
  "prefer-template",
  "prettier/prettier",
  "quote-props",
  "quotes",
  "radix",
  "require-await",
  "require-jsdoc",
  "require-yield",
  "rest-spread-spacing",
  "semi",
  "semi-spacing",
  "semi-style",
  "sort-imports",
  "sort-keys",
  "sort-vars",
  "space-before-blocks",
  "space-before-function-paren",
  "space-in-parens",
  "space-infix-ops",
  "space-unary-ops",
  "spaced-comment",
  "strict",
  "switch-colon-spacing",
  "symbol-description",
  "template-curly-spacing",
  "template-tag-spacing",
  "unicode-bom",
  "use-isnan",
  "valid-jsdoc",
  "valid-typeof",
  "vars-on-top",
  "wrap-iife",
  "wrap-regex",
  "yield-star-spacing",
  "yoda",
]
`;
